packages/hardhat-core/src/internal/core/config/config-loading.ts

```diff
@@ -34,9 +34,13 @@ export function importCsjOrEsModule(filePath: string): any {
     // The kind of error we get in these cases depends on the Node.js version.
     const node20Heuristic = e.code === "ERR_REQUIRE_ESM";
     // Also works for node v24
+    const lines = e.message.split("\n");
+    const message = lines.length > 0 ? lines[0] : "";
```

Comment:
Are all of the messages we chech against in the heuristics expected to be multiline (hence the `""` default)?